woocommerce/woocommerce-gutenberg-products-block
# Fix an issue with COD not showing when first enabled

This PR fixes 3087 by defaulting to an array if `enable_for_methods` is not set.

The issue here is that WooCommerce will set `enable_for_methods` as an empty string when interacting with the field, but if you don't touch it, it won't be set, and it would default to an empty array.
This also changes the logic of COD `canMakePayment` to account for arrays, previously it assumed that `""` was the only empty value, but `[]` could also mean we don't have any values, doing `! settings.enableForShippingMethods` is not enough because `[]` is a truthy value, you can check this for examples about truthy values (#2024).

The original steams from WooCommerce and how it changes value types between states (switching from an array to an empty string).


Fixes #3087


### How to test the changes in this Pull Request:


1. You can either test this on a new website, or delete `woocommerce_cod_settings` option from `wp_options` table.
2. Enable Cash on Delivery, don't set it up, leave everything as is, it should work with all shipping methods.
3. Create an order and enter a valid address that has a shipping method with it.
4. You will see COD as a payment option.
5. Go to COD settings, select a value for "Enable for shipping methods".
6. Save the settings.
7. Delete what select to return the field to its initial state.
8. Visit checkout again, COD is still working.
